[@types/sharp] "sharp.resize()" do not accept a object of options.

If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @lith-light-g @wooseopkim @BTOdell @JamieWoodbury 

Sharp [documentation example](https://sharp.pixelplumbing.com/en/stable/api-resize/#resize):
```js
sharp(input)
  .resize({ width: 100 })
  .toBuffer()
  .then(data => {
    // 100 pixels wide, auto-scaled height
  });
```

Current result:
```
error TS2345: Argument of type '{ width: number; }' is not assignable to parameter of type 'number'.
```

--- a/types/sharp/index.d.ts
+++ b/types/sharp/index.d.ts
@@ -1,4 +1,4 @@
-// Type definitions for sharp 0.23.3
+// Type definitions for sharp 0.23
 // Project: https://github.com/lovell/sharp
 // Definitions by: François Nguyen <https://github.com/lith-light-g>
 //                 Wooseop Kim <https://github.com/wooseopkim>

--- a/types/sharp/index.d.ts
+++ b/types/sharp/index.d.ts
@@ -1,4 +1,4 @@
-// Type definitions for sharp 0.23
+// Type definitions for sharp 0.23.3
 // Project: https://github.com/lovell/sharp
 // Definitions by: François Nguyen <https://github.com/lith-light-g>
 //                 Wooseop Kim <https://github.com/wooseopkim>
@@ -586,6 +586,15 @@ declare namespace sharp {
         resize(width?: number | null, height?: number | null, options?: ResizeOptions): Sharp;
 
         /**
+         * Shorthand for resize(null, null, options);
+         *
+         * @param options resize options
+         * @throws {Error} Invalid parameters
+         * @returns A sharp instance that can be used to chain operations
+         */
+        resize(options: ResizeOptions): Sharp;
+
+        /**
          * Extends/pads the edges of the image with the provided background colour.
          * This operation will always occur after resizing and extraction, if any.
          * @param extend single pixel count to add to all edges or an Object with per-edge counts
@@ -929,6 +938,8 @@ declare namespace sharp {
         overlap?: number;
         /** Tile angle of rotation, must be a multiple of 90. (optional, default 0) */
         angle?: number;
+        /** background colour, parsed by the color module, defaults to white without transparency. (optional, default {r:255,g:255,b:255,alpha:1}) */
+        background?: string | RGBA;
         /** How deep to make the pyramid, possible values are "onepixel", "onetile" or "one" (default based on layout) */
         depth?: string;
         /** Threshold to skip tile generation, a value 0 - 255 for 8-bit images or 0 - 65535 for 16-bit images */

--- a/types/sharp/sharp-tests.ts
+++ b/types/sharp/sharp-tests.ts
@@ -46,6 +46,10 @@ sharp('input.jpg')
         // containing a scaled and cropped version of input.jpg
     });
 
+sharp('input.jpg')
+    .resize({width: 300})
+    .toFile('output.jpg');
+
 sharp({
     create: {
         width: 300,

